lib/array.rb

```diff
@@ -0,0 +1,13 @@
+class Array
+  def my_inject *args, &block
+    result = args.detect { |argument| !(argument.is_a?(Symbol)) } || shift
+    symbol = args.detect { |argument| argument.is_a?(Symbol) }
+    each { |elem| result = block.call(result, elem) } unless symbol
+    if [:*, :/, :+, :-].include?(symbol)
+      each { |elem| result = result.send(symbol, elem) }
```

Comment:
This is nice. To make it clearer that you were only going to support these methods, maybe extract `[:*, :/, :+, :-]` to a separate `MY_INJECT_SUPPORTED_OPERATIONS` constant